src/LiveQuery/ParseLiveQueryServer.js

```diff
@@ -327,6 +336,52 @@ class ParseLiveQueryServer {
     return matchesQuery(parseObject, subscription.query);
   }
 
+  _matchesCLP(classLevelPermissions: ?any, object: any, client: any, requestId: number, op: string): any {
+    return Parse.Promise.as().then(() => {
+      // try to match on user first, less expensive than with roles
+      const subscriptionInfo = client.getSubscriptionInfo(requestId);
+      if (typeof subscriptionInfo === 'undefined') {
+        return Parse.Promise.as(['*']);
+      }
+      let foundUserId;
+      const subscriptionSessionToken = subscriptionInfo.sessionToken;
+      return this.sessionTokenCache.getUserId(subscriptionSessionToken).then((userId) => {
+        foundUserId = userId;
+        if (userId) {
+          return Parse.Promise.as(['*', userId]);
+        }
+        return Parse.Promise.as(['*']);
+      }).then((aclGroup) => {
+        console.log(aclGroup); // eslint-disable-line
+        try {
+          return SchemaController.validatePermission(classLevelPermissions, object.className, aclGroup, op).then(() => {
+            return Parse.Promise.as(true);
+          });
+        } catch(e) {
+          logger.verbose(`Failed matching CLP for ${object.id} ${foundUserId} ${e}`);
+          return Parse.Promise.as(false);
+        }
+        // TODO: handle roles permissions
```

Comment:
we can probably let those out for now, because

- nothing was supported before
- The default behaviour is the safe one.
- roles require more i/o / query, and could be cached properly

Reply:
Hmmm, I don't know about this one. Having the live query responses behave differently in terms of roles could cause some unexpected headaches :/.

Reply:
An alternative could be a cache of roles?

Reply:
yep, we'd need a roles cache somehow, perhaps refactoring the Auth module.

Reply:
Refactoring is finally done in #4940 :)